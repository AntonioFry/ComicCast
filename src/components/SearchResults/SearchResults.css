.search-results {
  display: grid;
  grid-gap: 20px 0px;
  grid-template-columns: repeat(4, 1fr);
  padding: 30px;
}