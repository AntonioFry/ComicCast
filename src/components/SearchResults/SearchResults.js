import React from 'react';
import Card from '../Card/Card';
import './SearchResults.css'
import { connect } from 'react-redux';

export const SearchResults = ({ allCharacters, search }) => {
  const filteredCharacters = allCharacters.filter(character => {
    return character.name.toLowerCase().includes(search);
  });
  const characterCards = filteredCharacters.map(character => {
    return (
      <Card
        name={character.name}
        id={character.id}
        image={character.image}
        key={character.id}
      />
    )
  });
  return (
    <section className="search-results">
      {characterCards}
    </section>
  )
}

const mapStateToProps = (store) => ({
  allCharacters: store.allCharacters,
  search: store.search
});

export default connect(mapStateToProps)(SearchResults);