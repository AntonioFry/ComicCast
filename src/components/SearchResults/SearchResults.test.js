import React from 'react';
import { SearchResults } from './SearchResults';
import { shallow } from 'enzyme';

describe('SearchResults', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      allCharacters: [
        { name: 'guy', id: 1, image: 'image.jpg', key: 1 },
        { name: 'peep', id: 2, image: 'image.jpg', key: 2 },
        { name: 'gal', id: 3, image: 'image.jpg', key: 3 },
        { name: 'dude', id: 4, image: 'image.jpg', key: 4 }
      ]
    } 
    wrapper = shallow(<SearchResults allCharacters={props.allCharacters} search=""/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

});