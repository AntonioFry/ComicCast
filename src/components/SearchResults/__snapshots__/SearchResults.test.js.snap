// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult should match the snapshot 1`] = `undefined`;

exports[`SearchResults should match the snapshot 1`] = `undefined`;
